website_product_configurator/static/src/js/config_form.esm.js

```diff
@@ -0,0 +1,629 @@
+        "change #product_config_form .config_attribute": "_onChangeConfigAttribute",
+        "change #product_config_form .custom_config_value.config_attachment":
+            "_onChangeFile",
+        "change #product_config_form .custom_config_value": "_onChangeCustomField",
+        "click #product_config_form .config_step": "_onClickConfigStep",
+        "click #product_config_form .btnNextStep": "_onClickBtnNext",
+        "click #product_config_form .btnPreviousStep": "_onClickBtnPrevious",
+        "submit #product_config_form": "_onSubmitConfigForm",
+        "click #product_config_form .image_config_attr_value_radio":
+            "_onClickRadioImage",
+        "change #product_config_form .custom_config_value.spinner_qty":
+            "_onChangeQtySpinner",
+        "click #product_config_form .js_add_qty": "_onClickAddQty",
+        "click #product_config_form .js_remove_qty": "_onClickRemoveQty",
+    },
+
+    init: function () {
+        this._super.apply(this, arguments);
+        this.config_form = $("#product_config_form");
+        // For file (custom field)
+        this.image_dict = {};
+    },
+
+    _onChangeConfigAttribute: function (event) {
+        var self = this;
+        var attribute = [event.currentTarget];
+        self._checkRequiredFields(attribute);
+        var flag = self._checkChange(self);
+        if (flag) {
+            var form_data = self.config_form.serializeArray();
+            for (var field_name in self.image_dict) {
+                form_data.push({
+                    name: field_name,
+                    value: self.image_dict[field_name],
+                });
+            }
+
+            self.call("ui", "block");
+            jsonrpc("/website_product_configurator/onchange", {
+                form_values: form_data,
+                field_name: $(attribute[0]).getAttributes().name,
+            }).then(function (data) {
+                if (data.error) {
+                    self.openWarningDialog(data.error);
+                } else {
+                    var values = data.value;
+                    var domains = data.domain;
+
+                    var open_cfg_step_line_ids = data.open_cfg_step_line_ids;
+                    var config_image_vals = data.config_image_vals;
+
+                    self._applyDomainOnValues(domains);
+                    self._setDataOldValId();
+                    self._handleOpenSteps(open_cfg_step_line_ids);
+                    self._setImageUrl(config_image_vals);
+                    self._setWeightPrice(
+                        values.weight,
+                        values.price,
+                        data.decimal_precision
+                    );
+                }
+                self.call("ui", "unblock");
+            });
+            self._handleCustomAttribute(event);
+        }
+    },
+
+    _checkChange: function (attr_field) {
+        var flag = true;
+        if ($(attr_field).hasClass("cfg-radio")) {
```

Comment:
nitpicking / not-blocking:

use of jQuery in Odoo is deprecated.
Ideally, use vanillaJS

(here, and the rest of the code too)

e.g:

```js
// $(attr_field).hasClass("cfg-radio")
attr_field.classList.contains("cfg-radio")

// $(attr_field).attr("data-old-val-id")
attr_field.getAttribute("data-old-val-id")
// or...
attr_field.dataset.oldValId

// $(attr_field).val()
attr_field.value
```